runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -81,7 +82,8 @@ public final class Engine implements AutoCloseable
     private final Collection<AgentRunner> runners;
     private final Tuning tuning;
     private final List<EngineExtSpi> extensions;
-    private final EngineExtContext context;
+    // visible for testing
+    final EngineExtContext context;
```

Comment:
Please keep this `private` and add package-private accessor method for test purposes instead.